common/primitives/timestamp/time.go

```diff
@@ -214,3 +214,15 @@ func UnixOrZeroTime(nanos int64) time.Time {
 
 	return time.Unix(0, nanos)
 }
+
+func UnixOrZeroTimePtr(nano int64) *time.Time {
+	return TimePtr(UnixOrZeroTime(nano))
+}
+
+func TimeNowPtrUtcAddSeconds(seconds int) *time.Time {
+	return TimePtr(time.Now().UTC().Add(time.Second * time.Duration(seconds)))
+}
```

Comment:
Subjectively, I am against such helpers because they makes code harder to read. I would use minimum set of primitives such as `TimePtr`/`TimeValue` and do all the rest in place.

Reply:
I am not saying you need to remove it, of course.

Reply:
Lets chat on this and come back and collapse these into a common set for a shared util package.